R/subject_industry.R

```diff
@@ -1,7 +1,962 @@
+      arrange(-All) %>%
+      mutate_at(vars(-group_cols()), funs(ifelse(is.na(.), 0, .))) %>%
+      mutate_at(vars(-group_cols()), funs(ifelse(. <= 2, 0, .))) %>%
+      mutate_if(is.numeric, funs(. / sum(.))) %>%
+      mutate_at(vars(-group_cols()), funs(ifelse(. == 0, NA, .))) %>%
+      mutate_at(
+        c(
+          "North East", "North West", "Yorkshire and the Humber", "East Midlands", "West Midlands",
+          "East of England", "London", "South East", "South West"
+        ),
+        funs(as.numeric(.))
+      ) %>%
+      # We can show all regions (including Abroad, Scotland, Wales and Northern Ireland) if we want too.
+      select(
+        SECTIONNAME, `North East`, `North West`, `Yorkshire and the Humber`, `East Midlands`, `West Midlands`,
+        `East of England`, `London`, `South East`, `South West`
+      )
+
+    crosstabs_earnings_data <- tables_data_grouped %>%
+      select(-count, n = earnings_median) %>%
+      spread(current_region, n) %>%
+      arrange(-All) %>%
+      mutate_at(vars(-group_cols()), funs(ifelse(is.na(.), 0, .))) %>%
+      mutate_at(vars(-group_cols()), funs(ifelse(. == 0, NA, .))) %>%
+      mutate_at(
+        c(
+          "North East", "North West", "Yorkshire and the Humber", "East Midlands", "West Midlands",
+          "East of England", "London", "South East", "South West"
+        ),
+        funs(as.numeric(.))
+      ) %>%
+      mutate_at(vars(-group_cols()), funs(ifelse(!is.na(as.numeric(.)), round(as.numeric(.), -2), .))) %>%
+      # We can show all regions (including Abroad, Scotland, Wales and Northern Ireland) if we want too.
+      select(
+        SECTIONNAME, `North East`, `North West`, `Yorkshire and the Humber`, `East Midlands`, `West Midlands`,
+        `East of England`, `London`, `South East`, `South West`
+      )
+
+    regionfirst <- function(current_region) {
+      first(crosstabs_data$SECTIONNAME, order_by = -crosstabs_data[current_region])
+    }
+
+    regionfirstdata <- c(
+      regionfirst("North East"), regionfirst("North West"), regionfirst("Yorkshire and the Humber"), regionfirst("East Midlands"),
+      regionfirst("West Midlands"), regionfirst("East of England"), regionfirst("London"), regionfirst("South East"),
+      regionfirst("South West")
+    )
+    regionfirstdata <- data.frame(regionfirstdata)
+    regionfirstdata$region <- c(
+      "the North East", "the North West", "Yorkshire and the Humber", "the East Midlands", "the West Midlands",
+      " the East of England", "London", "the South East", "the South West"
+    )
+
+    uniqueregions <- unique(regionfirstdata$regionfirstdata)
+
+    textprod <- function(data) {
+      if (length(data) != 1) {
+        x <- paste(data$region[1:nrow(data) - 1], collapse = ", ")
+        y <- paste(" and ", data$region[nrow(data)], sep = "")
+        paste(x, y)
+      } else if (length(data) == 1) {
+        paste(data$region[1])
+      }
+    }
+
+
+    print(length(uniqueregions))
```

Comment:
Don't think this print() is needed - were you using it while developing?

Reply:
Guessing I must have been at some point, will delete it.